lib/ansible/modules/storage/hpe3par/cpg_3par.py

```diff
@@ -0,0 +1,322 @@
+            ld_layout = cpg_ldlayout_map(ld_layout)
+            if growth_increment is not None:
+                growth_increment = convert_to_binary_multiple(
+                    growth_increment)
+            if growth_limit is not None:
+                growth_limit = convert_to_binary_multiple(
+                    growth_limit)
+            if growth_warning is not None:
+                growth_warning = convert_to_binary_multiple(
+                    growth_warning)
+            optional = {
+                'domain': domain,
+                'growthIncrementMiB': growth_increment,
+                'growthLimitMiB': growth_limit,
+                'usedLDWarningAlertMiB': growth_warning,
+                'LDLayout': ld_layout}
+            client_obj.createCPG(cpg_name, optional)
+        else:
+            return (True, False, "CPG already present")
+    except exceptions.ClientException as e:
+        return (False, False, "CPG creation failed | %s" % (e))
+    return (True, True, "Created CPG %s successfully." % cpg_name)
+
+
+def delete_cpg(
+        client_obj,
+        cpg_name):
+    try:
+        if client_obj.cpgExists(cpg_name):
+            client_obj.deleteCPG(cpg_name)
+        else:
+            return (True, False, "CPG does not exist")
+    except exceptions.ClientException as e:
+        return (False, False, "CPG delete failed | %s" % e)
+    return (True, True, "Deleted CPG %s successfully." % cpg_name)
+
+
+def main():
+    module = AnsibleModule(argument_spec=hpe3par.cpg_argument_spec())
+    if not HAS_3PARCLIENT:
+        module.fail_json(msg='the python hpe3par_sdk library is required (https://pypi.org/project/hpe3par_sdk)')
+
+    if len(module.params["cpg_name"]) < 1 or len(module.params["cpg_name"]) > 31:
+        module.fail_json(msg="CPG name must be atleast 1 character and not more than 31 characters")
+
+    storage_system_ip = module.params["storage_system_ip"]
+    storage_system_username = module.params["storage_system_username"]
+    storage_system_password = module.params["storage_system_password"]
+    cpg_name = module.params["cpg_name"]
+    domain = module.params["domain"]
+    growth_increment = module.params["growth_increment"]
+    growth_limit = module.params["growth_limit"]
+    growth_warning = module.params["growth_warning"]
+    raid_type = module.params["raid_type"]
+    set_size = module.params["set_size"]
+    high_availability = module.params["high_availability"]
+    disk_type = module.params["disk_type"]
+    secure = module.params["secure"]
+
+    wsapi_url = 'https://%s:8080/api/v1' % storage_system_ip
+    try:
+        client_obj = client.HPE3ParClient(wsapi_url, secure)
+    except exceptions.SSLCertFailed:
+        module.fail_json(msg="SSL Certificate Failed")
+    except exceptions.ConnectionError:
+        module.fail_json(msg="COnnection Error")
+    except exceptions.UnsupportedVersion:
+        module.fail_json(msg="Unsupported WSAPI version")
+    except:
```

Comment:
Bare `except` should be avoided ([see](https://docs.python.org/2/howto/doanddont.html#except)). It might be helpful to include exception in `msg`.